kart/tabular/export.py

```diff
@@ -0,0 +1,306 @@
+
+    ds_path = args[0]
+    destination_spec = args[1]
+
+    try:
+        dataset = repo.datasets(ref)[ds_path]
+    except KeyError:
+        raise click.UsageError(f"No such dataset: {ds_path}")
+
+    if dataset.DATASET_TYPE != "table":
+        raise click.UsageError(f"Dataset {ds_path} is not a vector or tabular dataset")
+
+    driver, destination = get_driver(destination_spec)
+    out_ds = driver.CreateDataSource(destination, options=list(dataset_creation_option))
+    layer_name = layer or ds_path
+
+    if override_geometry and override_geometry.upper() not in OGR_GEOMTYPE_MAP:
+        click.echo("Geometry type should be one of:", err=True)
+        click.echo("\n".join(OGR_GEOMTYPE_MAP.keys()), err=True)
+        raise click.UsageError(f"Unknown geometry type: {override_geometry}")
+
+    schema = dataset.schema
+    pk_name = schema.first_pk_column.name
+
+    geom_key = None
+    ogr_geom_type = None
+    ogr_crs = None
+
+    if schema.has_geometry:
+        geom_col = schema.geometry_columns[0]
+        geom_key = geom_col.name
+        kart_geom_type = override_geometry or geom_col["geometryType"]
+        ogr_geom_type = OGR_GEOMTYPE_MAP[kart_geom_type.upper()]
+
+        crs_defs = list(dataset.crs_definitions().values())
+        if len(crs_defs) > 1:
+            raise CrsError(
+                f"Sorry, multiple CRS definitions at {ds_path!r} are not yet supported for export"
+            )
+        if crs_defs:
+            ogr_crs = make_crs(crs_defs[0], context=ds_path)
+
+        out_layer = out_ds.CreateLayer(
+            layer_name, ogr_crs, ogr_geom_type, options=list(layer_creation_option)
+        )
+    else:
+        out_layer = out_ds.CreateLayer(layer_name, options=list(layer_creation_option))
+
+    regular_keys = []
+    for col in dataset.schema:
+        if col.name == pk_name and not primary_key_as_field:
+            continue
+        if col.data_type == "geometry":
+            continue  # Handled separately
+        regular_keys.append(col.name)
+        out_layer.CreateField(kart_col_to_ogr_field(col))
+
+    for feature in dataset.features_with_crs_ids(
+        repo.spatial_filter, show_progress=True
+    ):
+        out_feature = ogr.Feature(out_layer.GetLayerDefn())
+        if primary_key_as_fid:
+            out_feature.SetFID(feature[pk_name])
+        if geom_key:
+            geom = feature[geom_key]
+            out_feature.SetGeometry(geom.to_ogr() if geom else None)
+        for i, key in enumerate(regular_keys):
+            out_feature.SetField(i, feature[key])
+        out_layer.CreateFeature(out_feature)
+        out_feature = None
+
+    out_ds = None
+
+
+def kart_col_to_ogr_field(col):
+    ogr_type = None
+    size = col.get("size")
+    if size is not None:
+        ogr_type = OGR_DATATYPE_MAP.get((col.data_type, size))
+    if ogr_type is None:
+        ogr_type = OGR_DATATYPE_MAP[col.data_type]
+    assert ogr_type is not None
+
+    if isinstance(ogr_type, tuple):
```

Comment:
just make all the values in OGR_DATATYPE_MAP tuples with subtypes None where there's no subtype